packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -142,7 +144,21 @@ class Dialog extends AutoControlledComponent<WithAsProp<DialogProps>, DialogStat
   triggerRef = React.createRef<HTMLElement>()
 
   getInitialAutoControlledState(): DialogState {
-    return { open: false }
+    return {
+      contentId: _.uniqueId('dialog-content-'),
+      headerId: _.uniqueId('dialog-header-'),
+      open: false,
+    }
+  }
+
+  static getAutoControlledStateFromProps(
+    props: DialogProps,
+    state: DialogState,
+  ): Partial<DialogState> {
+    return {
+      contentId: (props.content && props.content['id']) || state.contentId,
```

Comment:
```
props.content && props.content['id']
```

what if React element is passed to any of these slots, with `id` provided? As I see, in that case we will silently override client's ID value - this is definitely what client expects in this situation. 

As we've committed to handle the props calculation for React elements provided to slots, we should handle this case as well.

Reply:
Good catch, will rework it